common/types/pb/type.go

```diff
@@ -18,23 +18,50 @@ import (
 	"fmt"
 	"reflect"
 	"strings"
-	"sync/atomic"
+	"sync"
 
 	"github.com/golang/protobuf/proto"
 	descpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
 	structpb "github.com/golang/protobuf/ptypes/struct"
 	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
 )
 
+// NewTypeDescription produces a TypeDescription value for the proto type name with a given
```

Comment:
Specify whether the type name is fully-qualified.

Reply:
Done.